ducktape/services/service.py

```diff
@@ -51,11 +52,17 @@ class Service(TemplateRenderer):
     # }
     logs = {}
 
-    def __init__(self, context, num_nodes, *args, **kwargs):
+    # TODO: I changed this from num_nodes to nodes but I may not have updated all places where this constructor is called.
```

Comment:
It's actually difficult to validate this since this is user facing code, right? One option to maintain compatibility would be to keep the num_nodes argument and convert it to the new node spec if it is used. It would be safe to convert it to `{'linux':num_nodes}` since Linux support was all that was previously provided.

Without the compatibility we'd probably actually want to bump the version # to 0.7 since it'd be an incompatible change. To maintain compatibility we'd probably want to add the `nodes` argument with a default value.